processing/src/main/java/org/apache/druid/query/expression/TimestampExtractExprMacro.java

```diff
@@ -71,8 +70,6 @@ private ExprEval getExprEval(final DateTime dateTime, final Unit unit)
     switch (unit) {
       case EPOCH:
         return ExprEval.of(epoch);
-      case MICROSECOND:
-        return ExprEval.of(epoch / 1000);
```

Comment:
Retaining it, the result will not be completely correct since the precision will be till ms only. Moreover for query like these will return 0 when ideally the result should be 100

SELECT (TIME_EXTRACT(CAST('2013-10-07 12:13:27.000000' AS TIMESTAMP), 'MICROSECOND') - TIME_EXTRACT(CAST('2013-10-07 12:13:27.000100' AS TIMESTAMP), 'MICROSECOND') )

Reply:
yes - it would be rounded...

because it was broken - I don't think anyone was using it at all....